react/src/views/Box/BoxView.tsx

```diff
@@ -169,6 +175,38 @@ function BTBox() {
     ? [BoxState.Receiving, BoxState.MarkedForShipment, BoxState.InTransit].includes(currentBoxState)
     : false;
 
+  // map over each box HistoryEntry to compile its timeline records
+  const boxLogs: ITimelineEntry[] = (allData.data?.box?.history as HistoryEntry[])?.flatMap(
+    (histories) =>
+      _.compact([
+        histories?.user && {
+          action: prepareBoxHistoryEntryText(`${histories.user.name} ${histories.changes}`),
+          createdBy: histories.user as User,
+          createdOn: new Date(histories.changeDate),
+        },
+      ]),
+  ) as unknown as ITimelineEntry[];
```

Comment:
could you try that the data that is returned here, does not include any unknowns, undefined or similar. As mentioned above it makes the definition of the UI component cleaner and easier.